virttest/utils_net.py

```diff
@@ -3008,15 +3008,25 @@ def get_ip_address_by_interface(ifname):
 
 def get_host_ip_address(params):
     """
-    returns ip address of host specified in host_ip_addr parameter If provided
-    otherwise ip address on interface specified in netdst parameter is returned
+    Returns ip address of host specified in host_ip_addr parameter if provided.
+    Otherwise ip address on interface specified in netdst parameter is returned.
+    In case of "nettype == user" "netdst" parameter is left empty, then the
+    first interface of the system is used.
     :param params
     """
     host_ip = params.get('host_ip_addr', None)
-    if not host_ip:
-        host_ip = get_ip_address_by_interface(params.get('netdst'))
-        logging.warning("No IP address of host was provided, using IP address"
-                        " on %s interface", str(params.get('netdst')))
+    if host_ip:
+        logging.debug("Use IP address at config %s=%s", 'host_ip_addr', host_ip)
+        return host_ip
+    net_dev = params.get('netdst')
+    if not net_dev:
+        net_dev = get_host_iface()
+    if not net_dev:
+        raise NetError("Error while retrieving IP address for host.")
+    net_dev = net_dev[0]
```

Comment:
Agree, the first dev may not the one we want, check out this case:
```
 # cat /proc/net/dev 
Inter-|   Receive                                                |  Transmit
 face |bytes    packets errs drop fifo frame compressed multicast|bytes    packets errs drop fifo colls carrier compressed
macvtap2: 31110323826 211800342    0    0    0     0          0   8098497 14559465793 98202079    0  258    0     0       0          0
macvtap3: 473310902 3355236    0    0    0     0          0   1577470  6216236   68602    0    0    0     0       0          0
macvtap4: 387571748 2679736    0    0    0     0          0   1271033  4987953   59501    0    0    0     0       0          0
    lo: 136565106974 71946861    0    0    0     0          0         0 136565106974 71946861    0    0    0     0       0          0
virbr0-nic:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0
virbr0:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0
   em2:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0
   em1: 20600661130 42322580    0 101698    0     0          0  19625486 6970419340 9894755    0    0    0     0       0          0
   em4:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0
   em3:       0       0    0    0    0     0          0         0        0       0    0    0    0     0       0          0
macvtap0: 30359412718 211395929    0    0    0     0          0   8126047 13239372372 97375244    0  112    0     0       0          0
macvtap1: 30856475158 224759945    0    0    0     0          0   8073646 17325696474 105454812    0  120    0     0       0          0
```
Here, `em1` should be the interface to reach out